src/common/snippets/include/snippets/lowered/port_descriptor.hpp

```diff
@@ -43,7 +43,11 @@ class PortDescriptor {
 
     void set_shape(const VectorDims& tensor);
     void set_layout(const std::vector<size_t>& layout) { m_layout = layout; }
-    void set_subtensor(const VectorDims& subtensor) { m_subtensor_shape = subtensor; }
+    void set_subtensor(const VectorDims& subtensor) {
+        OPENVINO_ASSERT(m_subtensor_shape.size() <= m_tensor_shape->size(),
+                        "Subtensor shape must be less than or equal to tensor shape");
+        m_subtensor_shape = subtensor;
+    }
```

Comment:
Since we add a check in this setter, shouldn't we update `set_layout` `set_shape` for consistency? In `set_shape`, we could check
```
    OPENVINO_ASSERT(m_subtensor_shape.size() <= m_tensor_shape->size(),
                    "Snippets tensor descriptor: Subtensor shape must be less than or equal to tensor shape");
    OPENVINO_ASSERT(m_layout.size() == m_tensor_shape->size(), "Snippets tensor descriptor: Layout size must be equal to the shape size");
```
In `set_layout`, only `m_layout.size() == m_tensor_shape->size()` is valid. What do you think?

Reply:
Agree, applied

Reply:
`set_shape` is still not updated, please take a look